lib/pharos/cluster_manager.rb

```diff
@@ -132,7 +132,16 @@ def apply_phases
       apply_phase(Phases::ConfigureTelemetry, [master_hosts.first], master: master_hosts.first)
     end
 
-    def apply_reset
+    def apply_reset_hosts(hosts)
+      master_hosts = sorted_master_hosts
+      apply_phase(Phases::GatherFacts, hosts, ssh: true, parallel: true)
+      apply_phase(Phases::ConfigureClient, [master_hosts.first], ssh: true, master: master_hosts.first, parallel: false, optional: true)
+      apply_phase(Phases::Drain, hosts, parallel: false)
```

Comment:
Drain & DeleteHost should be optional.. probably also ConfigureClient because master might be down.